packages/test-sync/src/androidAndroidTest/kotlin/io/realm/kotlin/test/mongodb/shared/FunctionsTests.kt

```diff
@@ -0,0 +1,623 @@
+                    )
+                    assertEquals(
+                        32L,
+                        functions.callBlocking<Long>(FIRST_ARG_FUNCTION.name, 32).toLong()
+                    )
+                    assertTypeOfFirstArgFunction(BsonInt64(32))
+                }
+                BsonType.DECIMAL128 -> {
+                    assertTypeOfFirstArgFunction(BsonDecimal128("32"))
+                }
+                BsonType.DOCUMENT -> {
+                    val map = mapOf("foo" to 5)
+                    val document = BsonDocument(mapOf("foo" to BsonInt32(5)))
+
+                    assertEquals(
+                        document,
+                        functions.callBlocking<BsonDocument>(FIRST_ARG_FUNCTION.name, map)
+                    )
+                    assertEquals(
+                        document,
+                        functions.callBlocking<BsonDocument>(FIRST_ARG_FUNCTION.name, document)
+                    )
+
+                    var documents = arrayOf(BsonDocument(), BsonDocument())
+                    assertEquals(
+                        documents[0],
+                        functions.callBlocking<BsonDocument>(FIRST_ARG_FUNCTION.name, *documents)
+                    )
+
+                    documents = arrayOf(
+                        BsonDocument("KEY", BsonString("VALUE")),
+                        BsonDocument("KEY", BsonString("VALUE")),
+                        BsonDocument("KEY", BsonString("VALUE"))
+                    )
+                    assertEquals(
+                        documents[0],
+                        functions.callBlocking<BsonDocument>(FIRST_ARG_FUNCTION.name, *documents)
+                    )
+                }
+                BsonType.DATE_TIME -> {
+                    // JVM and Darwing platform's RealmInstant have better precission than BsonDateTime
+                    // we create a RealmInstant with loose of precision
+                    val nowWithPrecisionLoose = RealmInstant.now().toMillis()
+                    val now = nowWithPrecisionLoose.toRealmInstant()
+
+                    assertEquals(
+                        now,
+                        functions.callBlocking<RealmInstant>(FIRST_ARG_FUNCTION.name, now)
+                    )
+                }
+                BsonType.UNDEFINED,
+                BsonType.NULL -> {
+                    assertNull(functions.callBlocking(FIRST_ARG_FUNCTION.name, null))
+                }
+                BsonType.REGULAR_EXPRESSION,
+                BsonType.SYMBOL,
+                BsonType.DB_POINTER,
+                BsonType.JAVASCRIPT,
+                BsonType.JAVASCRIPT_WITH_SCOPE,
+                BsonType.TIMESTAMP,
+                BsonType.END_OF_DOCUMENT,
+                BsonType.MIN_KEY,
+                BsonType.MAX_KEY -> {
+                    // Relying on org.bson codec providers for conversion, so skipping explicit
```

Comment:
Is there an issue for this that we can reference?

Reply:
I copied it from Java but you are right that its worth having them roundtripped.